### DISCUSSION TITLE
bipartite/multiple nodes update for undirected edge

### DISCUSSION THREAD

**Discussion by hg7661** [pyg-team/pytorch_geometric#7410] [Q&A]
With the standard MessagePassing class, it seems that node update is limited to either "source to target" or "source to target."

Given a bipartite graph, I would like to update nodes based on node and edge attributes, leaving edge attributes throughout the iteration/learning as constants. The two nodes are specified by the edge that connects those. The learning objective would be node regression.

In short, is there any base class that could be extended to update node i and node j  based on node i, node j, and edge ij?

**Comment by wsad1** [comment]
> In short, is there any base class that could be extended to update node i and node j based on node i, node j, and edge ij?\

The `MessagePassing` class always from sends messages source to target or vice-versa. So you could add edges in the opposite direction `edge ji` which is the `transpose` of `edge ij`. Then you'll have messages going in either direction.
